rotkehlchen/chain/bitcoin/__init__.py

```diff
@@ -13,53 +13,98 @@ def _have_bc1_accounts(accounts: List[BTCAddress]) -> bool:
     return any(account.lower()[0:3] == 'bc1' for account in accounts)
 
 
-def get_bitcoin_addresses_balances(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
-    """Queries blockchain.info or blockstream for the balances of accounts
+def _query_blockstream_info(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
+    """Queries balances from blockstream.info
+    May raise:
+    - RemoteError if got problems with querying the API
+    - KeyError if got problems with reading the returned json
+    """
+    balances = {}
+    for account in accounts:
+        url = f'https://blockstream.info/api/address/{account}'
+        response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
+        stats = response_data['chain_stats']
+        balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
+        balances[account] = satoshis_to_btc(balance)
+    return balances
+
+
+def _query_mempool_space(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
+    """Queries balances from mempool.space
+    May raise:
+    - RemoteError if got problems with querying the API
+    - KeyError if got problems with reading the returned json
+    """
+    balances = {}
+    for account in accounts:
+        url = f'https://mempool.space/api/address/{account}'
+        response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
+        stats = response_data['chain_stats']
+        balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
```

Comment:
This can also raise `ValueError` if either `stats['funded_txo_sum']` or the other one are non integers. I suggest using `deserialize_int_from_str` and catch the potential `DeserializationError` somewhere and handle it. Probably out of this function since this seems to be the kind of function that would let the errors propagate outwards.

If you let it propagate make sure it's mentioned in the docstring.